### PR TITLE
DBZ-1371 Fix tests to run more reliably on Amazon RDS

### PR DESCRIPTION
**Pull request by Naros**
https://issues.jboss.org/browse/DBZ-1371

### CODE REVIEW

**Comment by jpechane** [comment]
@Naros Thanks for the PR. I've two comments though

- I don't think the defautl timeouts should be changed as they are configurable via system property so this should be just a parameter to execution
- regarding other timeout changes - I think they should not be hardcoded but derived from the main configurable timeout - either multiplied or dvidied by some factor

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 1]
```diff
@@ -305,86 +305,96 @@ public void shouldConsumeMessagesFromSnapshot() throws Exception {
...
+            TestHelper.execute(
+                    "ALTER TABLE changepk.test_table DROP CONSTRAINT test_table_pkey;"
+                            + "ALTER TABLE changepk.test_table DROP COLUMN pk2;"
+                            + "ALTER TABLE changepk.test_table ADD COLUMN pk3 SERIAL;"
+                            + "ALTER TABLE changepk.test_table ADD PRIMARY KEY(newpk,pk3);"
+                            + "INSERT INTO changepk.test_table VALUES(5, 'dropandaddpkcol',10)" );
+
+            start( PostgresConnector.class, config.getConfig() );
+
+            records = consumeRecordsByTopic( 2 );
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java]: Minor nit: could you revert those whitespace changes? Specifically, we don't use the whitespace around parameter arguments. It looks like you applied the Hibernate template perhaps? We ought to expand CheckStyle for that, too.

File: debezium-connector-postgres/RDS.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+In order to run the test suite against an Amazon RDS PostgreSQL environment, there are several important steps that must be done in order for Debezium features to operate correctly on RDS and to allow remote connectivity to your Amazon Virtual Private Cloud (VPC) environment.
+
+## Setup remote connectivity
+
+When you setup an account with Amazon RDS, they should automatically create a new Security Group for your VPC environment.  By default this new security group does not allow remote connections from your local machine to the resources which it secures, so you you need to explicitly grant remote access.
```
Comment (gunnarmorling) [inline debezium-connector-postgres/RDS.md]: Could you use the "one sentence / unit-of-thought per line style". This avoids super-long lines while also not having to do hard breaks after a fixed number of characters. Diffs will be easier to view down the road that way.

File: debezium-connector-postgres/RDS.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+1. Navigate to **RDS**, this should take you to the RDS Dashbaord.
+2. Click **DB Instances** in the **Resources** panel on the dashboard.
+3. Click **Create database**.
+
+This is where all the _magic_ happens where we put tie together all the pieces of the prior configuration/setup we've done and how that interacts with our new RDS PostgreSQL environment.
+
+1. Click the radio button to select **PostgreSQL**
+2. Select the **Version** you wish to install.
+Remember this version should ideally match the version of database parameters you cloned in the prior steps.
```
Comment (gunnarmorling) [inline debezium-connector-postgres/RDS.md]:
```suggestion
Remember this version should match the version of database parameters you cloned in the prior steps.
```

File: debezium-connector-postgres/RDS.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+## Connecting to Amazon RDS
+
+In order to connect to your Amazon RDS PostgreSQL instance, you'll need to use the fully qualified endpoint name.  In order to determine your endpoint:
+
+1. Navigate to **RDS**, this should take you to the RDS Dashboard.
+2. Click **DB Instances** in the **Resources** panel on the dashboard.
+3. Click on the designed **DB Identifier** for the instance you're interested in connecting to.
+4. You should have a **Connectivity & security** tab where you can see the **Endpoint** name.
+
+When you run your JVM process you entire need to provide these VM options manually, provide them as a part of the exported environment or manually modify `TestHelper#defaultJdbcConfig()` specifying these values directly before running the tests.
```
Comment (gunnarmorling) [inline debezium-connector-postgres/RDS.md]:
```suggestion
When you run your JVM process you entirely need to provide these VM options manually, provide them as a part of the exported environment or manually modify `TestHelper#defaultJdbcConfig()` specifying these values directly before running the tests.
```

File: debezium-connector-postgres/RDS.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+1. Search for `log_replication_commands` and set its value to `1`.
+2. Search for `log_statement` and set its value to `all`.
+
+Once you've made these changes click `Save changes` to persist the parameter changes to your custom parameter group.
+
+## Setup database instance
+
+The latest RDS piece of setup we need to do is to actually install our desired PostgreSQL instance.
+
+1. Navigate to **RDS**, this should take you to the RDS Dashbaord.
```
Comment (gunnarmorling) [inline debezium-connector-postgres/RDS.md]:
```suggestion
1. Navigate to **RDS**, this should take you to the RDS Dashboard.
```

**Comment by Naros** [comment]
@jpechane I modified the ones I had originally changed to use the system property and multipled by a factor.  The only downside I have found with this is I have not yet found a way in IJ to configure a global runtime set of values I pass to all tests when running them as a package, but I can just use the command line approach in the meantime.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java [hunk 6]
```diff
@@ -234,8 +234,8 @@ protected static boolean shouldSSLConnectionFail() {
         return Boolean.parseBoolean(System.getProperty(TEST_PROPERTY_PREFIX + "ssl.failonconnect", "true"));
     }
 
-    protected static int waitTimeForRecords() {
-        return Integer.parseInt(System.getProperty(TEST_PROPERTY_PREFIX + "records.waittime", "2"));
+    public static int waitTimeForRecords() {
+        return Integer.parseInt(System.getProperty(TEST_PROPERTY_PREFIX + "records.waittime", "20"));
```
Comment (jpechane) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java]: @Naros Thanks for the timeout changes! The default should still be kept to the original value.

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java]: Also could you please undo the whitespace-only changes, @Naros!

**Comment by gunnarmorling** [comment]
@Naros, could you also add a link to the new instructions (which are super-helpful) from the main README (see https://github.com/debezium/debezium#running-tests-of-the-postgres-connector-against-an-external-database-eg-amazon-rds)?

File: debezium-connector-postgres/RDS.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+1. Navigate to **VPC** > **Security** > **Security Groups**
+2. Select your Security Group in the top pane
+3. Select the **Inbound Rules** tab in the bottom pane
+4. Select **Edit Rules**
+5. Add a new Rule
+    * Type: PostgreSQL
+    * Protocol: TCP
+    * Port Range: 5432
+    * Source: Anywhere
```
Comment (gunnarmorling) [inline debezium-connector-postgres/RDS.md]: How about this:

```suggestion
    * Source: 0.0.0.0/0 (for allowing access from any IP address), or a specific IP, host (range)
```

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java (hunk 6 above)
Comment (Naros) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java]: @gunnarmorling I can; I'm most likely going to need to reset IJ back to default and redo these format settings as its starting to be more and more problematic it seems :/

Comment (Naros) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java]: @gunnarmorling Are you referring to the changes in `PostgresConnectorIT`?

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/TestHelper.java]: @Naros I really meant all the changes in this PR which are only white-space changes (indentation). If I set this option in the PR view here on GH, most of the changes are hidden, i.e. the PR contains many superfluous changes. If it's too much of a hazzle to undo, let's talk.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks a lot, @Naros. I've pushed to more commits to revert the timeout changes and updating the README files some more.

**Pull request merged**